servicetalk-concurrent-api/docs/modules/ROOT/pages/blocking-implementation.adoc

```diff
@@ -38,22 +40,60 @@ Part 2. defines all the interactions using the ReactiveStreams specifications, i
+application performance by reducing latency and overhead. Requests to not offload will be honored by ServiceTalk if all
+the other components in the same execution path have also opted out of offloading. As a last resort, work tasks may also
+be queued to be performed as threads are available.
+
+ServiceTalk is designed to be fully asynchronous except where the API provides explicit blocking behaviour as a
+convenience. Offloading is the execution of program logic code using a thread (or equivalent) other than the application
+thread which originally initiated the operation. Offloading is used for two purposes within ServiceTalk; firstly for the
+execution needed for the handling of asynchronous events, aka Signals, and secondly when handling asynchronous events,
+protecting scarce resources from being monopolized by blocking, untrusted or expensive application code. Asynchronous
+execution requires offloading – the initiating application thread is not available. Protective offloading is a practical
+consideration, but just as necessary for reliable and predictable operation.
+
+Task based offloading uses an `Executor` in the traditional way to run the offloaded tasks. Often the Executor has a
+pool of threads, possibly unbounded, and tasks are run using whatever thread is available. In particular, different
+threads may be used for each task executed and code running in tasks cannot depend upon a consistent thread being used
+for invoking program logic. This approach is generally the most scalable because it makes the best utilization of
+threads.
+
+== Offloading and asynchronous sources
+
+ServiceTalk allows control of the thread that will be used for the signals of an asynchronous source. A safe
+used based upon the decision logic described earlier but for time–consuming tasks an `Executor` may be specified.
+The `subscribeOn(Executor)` and `publishOn(Executor)` operators will offload execution from the default thread to a
+thread from the provided `Executor`. The below diagram illustrates the interaction between an asynchronous source, its
+`Subscriber`, its operators, and the `Executor`.
+
+image::offloading.svg[Offloading]
+
+During `subscribe()` the execution will offload at the `subscribeOn()` operator and transition execution from the
+application thread to an `Exeuctor` thread. The application thread will be able to continue while the subscribe
+operation asynchronously continues on an `Executor` thread.
+
+When a result is available at the source it will begin
+publication using the receiving event loop thread but will offload at the `publishOn()` operator and transition
+execution from the event loop thread to an `Executor` thread. Once the publish signal is offloaded the event loop thread
+will be available again for executing other I/O tasks while the response is asynchronously processed on the `Executor`
+thread.
 
 Taking the same example from xref:{page-version}@servicetalk-concurrent-api::blocking-safe-by-default.adoc[here]
```

Comment:
`[here]` -> can you add an anchor link bcz this just goes to the top of the other page. also not sure why it needs to be a link if we are including the content again.

Reply:
also as described above the "executor affinity" concept doesn't apply any more so the example in `servicetalk-concurrent-api/blocking-safe-by-default.html` may go away ... if you think the example makes sense to keep in both places please make sure there is a meaningful difference in the content being discussed.